lib/httpoison/retry.ex

```diff
@@ -37,6 +37,7 @@ defmodule HTTPoison.Retry do
         max_attempts: Application.get_env(:httpoison_retry, :max_attempts) || unquote(@max_attempts),
         wait: Application.get_env(:httpoison_retry, :wait) || unquote(@reattempt_wait),
         include_404s: Application.get_env(:httpoison_retry, :include_404s) || false,
+        retry_all_errors: Application.get_env(:httpoison_retry, :retry_all_errors) || false,
```

Comment:
Maybe rename to `retry_unknown_errors` since thats what is actually happening here. Then default to `true` (you'll have to change the `|| false` to `Application.get_env(:httpoison_retry, :retry_unknown_errors, true)`).

Reply:
Hey Matt, I renamed the option to `retry_unknown_errors`. 

You mentioned defaulting to `true` for that option, but I was concerned that wouldn't be backwards compatible with the original retry settings and I just wanted to make sure that's what you meant?